crates/rattler_menuinst/src/lib.rs

```diff
@@ -0,0 +1,154 @@
+use std::path::Path;
+
+#[cfg(target_os = "linux")]
+use std::path::PathBuf;
+
+use rattler_conda_types::{
+    menuinst::{MenuMode, Tracker},
+    Platform,
+};
+
+#[cfg(target_os = "linux")]
+mod linux;
+#[cfg(target_os = "macos")]
+mod macos;
+mod render;
+mod schema;
+#[cfg(target_os = "windows")]
+mod windows;
+
+use crate::{render::BaseMenuItemPlaceholders, schema::MenuInstSchema};
+
+pub mod utils;
+
+#[derive(thiserror::Error, Debug)]
+pub enum MenuInstError {
+    #[error("IO error: {0}")]
+    IoError(#[from] std::io::Error),
+    #[error("Deserialization error: {0}")]
+    SerdeError(#[from] serde_json::Error),
+
+    #[error("Failed to install menu item: {0}")]
+    InstallError(String),
+
+    #[error("Invalid path: {0}")]
+    InvalidPath(PathBuf),
+
+    #[cfg(target_os = "linux")]
+    #[error("Could not quote command with shlex: {0}")]
+    ShlexQuoteError(#[from] shlex::QuoteError),
+
+    #[cfg(target_os = "macos")]
+    #[error("Failed to create plist: {0}")]
+    PlistError(#[from] plist::Error),
+    #[error("Failed to sign plist: {0}")]
+    SigningFailed(String),
+
+    #[error("Failed to install menu item: {0}")]
+    ActivationError(#[from] rattler_shell::activation::ActivationError),
+
+    #[cfg(target_os = "linux")]
+    #[error("Failed to install menu item: {0}")]
+    XmlError(#[from] quick_xml::Error),
+
+    #[cfg(target_os = "windows")]
+    #[error("Failed to install menu item: {0}")]
+    WindowsError(#[from] ::windows::core::Error),
+
+    #[cfg(target_os = "linux")]
+    #[error("Menu config location is not a file: {0:?}")]
+    MenuConfigNotAFile(PathBuf),
+}
```

Comment:
> The error message should be lowercase

From https://rust-lang.github.io/api-guidelines/interoperability.html?highlight=lower#error-types-are-meaningful-and-well-behaved-c-good-err

Reply:
fixed